File: median-of-two-sorted-arrays/median-of-two-sorted-arrays.rb
def find_median_sorted_arrays(nums1, nums2)
   median = 0.0
   nums1.concat nums2
   nums1.sort!
   len = nums1.length
   if (len % 2) == 0
        median = (nums1[len/2] + nums1[len/2 - 1]) / 2.0
   else
        median = nums1[(len-1)/2.0]
   end
end


# def find_median_sorted_arrays_logarithmic(nums1, nums2)
     
# end

File: median-of-two-sorted-arrays/median-of-two-sorted-arrays.go
package main

import (
	"fmt"
	"slices"
)

func findMedianSortedArrays(nums1 []int, nums2 []int) float64 {
	nums1 = slices.Concat(nums1, nums2)
	slices.Sort(nums1)
	length := len(nums1)
	var median float64

	if length%2 == 0 {
		median = float64(nums1[(length/2)-1]+nums1[length/2]) / 2
	} else {
		median = float64(nums1[(length-1)/2])
	}
	return median
}

func main() {
	// nums1 := []int{1, 2, 3}
	// nums2 := []int{4, 5}

	// nums1, nums2 := []int{1, 3}, []int{2}

	nums1 := []int{1, 2}
	nums2 := []int{3, 4}

	median := findMedianSortedArrays(nums1, nums2)
	fmt.Println(median)
}
